src/map/map_socket.h

```diff
@@ -37,7 +37,7 @@ class MapSocket
 public:
     using ReceiveFn = std::function<void(const std::error_code&, std::span<uint8>, IPP)>;
 
-    MapSocket(asio::io_context& io_context, uint16 port, const ReceiveFn& onReceiveFn); // TODO: Move passing in onReceiveFn to recvFor
+    MapSocket(asio::io_context& io_context, uint16 port, ReceiveFn onReceiveFn); // TODO: Move passing in onReceiveFn to recvFor
```

Comment:
There are two choices in cases like this (which the compiler can enforce and clang-tidy won't): 

We can use std::move_only_function as the type, or you can make it an r-value ref: `ReceiveFn&&`